### PR TITLE
Pure yarn

### PR DESCRIPTION
**Pull request by arun3528**
# COMPLETES # [SPARK-371483](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-371483)

## This pull request addresses

Upgraded most of the tooling to start using the yarn workspace , Going forward we just need to install yarn for everything from build to test'

Make changes to other tooling to make monorepo work as expected 

## by making the following changes

moved all the plugin to packages folder so that they can be handled independently 

ALL the tests failing on the branch are the same as the one failing in the master , Its been a while for some of the packages been updates so the test has not run for a while 

NOTE:

Search plugin testes are broken in master and are taking 30 + min then fails skipping it for now to fix this 
Screen Shot 2022-10-09 at 9 08 16 PM

From the testing performed locally here are the tests which are failing , They are failing on master as well and needs some attention 

This does not change any business logic so we are good , I compared and tested all the plugins and the results are attached in the JIRA 

```
@webex/http-core

SUMMARY:
✔ 86 tests completed
✖ 2 tests failed

internal-plugin-board

SUMMARY:
✔ 48 tests completed
ℹ 6 tests skipped
✖ 8 tests failed


@webex/internal-plugin-conversation

SUMMARY:
✔ 252 tests completed
ℹ 6 tests skipped
✖ 36 tests failed


PLugin device
SUMMARY:
✔ 110 tests completed
✖ 32 tests failed


plugin-encryption

SUMMARY:
✔ 84 tests completed
ℹ 4 tests skipped
✖ 10 tests failed

plugin-search

SUMMARY:
✔ 10 tests completed
✖ 14 tests failed

plugin-authorization-browser

SUMMARY:
✔ 0 tests completed
ℹ 2 tests skipped
✖ 4 tests failed

 plugin-messages

SUMMARY:
✔ 0 tests completed
✖ 4 tests failed

plugin-people ?
SUMMARY:
✔ 24 tests completed
✖ 2 tests failed

plugin-teams

SUMMARY:
✔ 14 tests completed
ℹ 2 tests skipped
✖ 2 tests failed

@webex/plugin-webhooks


SUMMARY:
✔ 0 tests completed
✖ 4 tests failed


webex 

SUMMARY:
✔ 14 tests completed
✖ 2 tests failed

Total 120 ✖ 
Total 696 ✔
```

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [x] Tooling change
- [x] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -11,10 +11,10 @@
     "WebSocket": false
   },
   "ignorePatterns": [
-    "!/**/node_modules/*",
```
[the remark this answers is not in the document]
Comment (arun3528) [reply .eslintrc.json:14]: It was running on it but not sure how , May be it was only ignoring the file and not folders inside the node_modules folder not sure 

I have anything PR with linting changes coming up after this

File: tooling/depcheck.mjs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+          let parsedPackages = JSON.parse(currentPackageJson);
+
+          let globalpackage = fs.readFileSync('globaldep.json');
+          let globalParsedPackages = JSON.parse(globalpackage);
+
+          let devDependencies = {}
+          let dependencies = {}
+          new Promise((res, reject) => {
+          fullList[path].using.forEach((dep, index) =>{
+            const key = `"`+dep+`"`;
```
[the remark this answers is not in the document]
Comment (arun3528) [reply tooling/depcheck.mjs]: created this to update the modules dont need it now

File: .eslintrc.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkesavan13) [reply .eslintrc.json:14]: Okay great then. Looks like eslint is running just now and hence we have those changes reflected in files. Resolving conversation.

**Review by InteractiveTimmy — COMMENTED** [review]
These are some huge changes in this PR around CD/CI that could impact versioning and deploy flows, as well as general build flows. I will check with the web client team to confirm if we are potentially okay with halting version bumps within our projects for up to two weeks to allow for a safe window to fix any potential issues that could arrise after merge.

**Pull request merged**